src/msw/webview_edge.cpp

```diff
@@ -0,0 +1,777 @@
+    m_historyLoadingFromList = true;
+    LoadURL(item->GetUrl());
+    m_historyPosition = pos;
+}
+
+wxVector<wxSharedPtr<wxWebViewHistoryItem> > wxWebViewEdge::GetBackwardHistory()
+{
+    wxVector<wxSharedPtr<wxWebViewHistoryItem> > backhist;
+    //As we don't have std::copy or an iterator constructor in the wxwidgets
+    //native vector we construct it by hand
+    for (int i = 0; i < m_historyPosition; i++)
+    {
+        backhist.push_back(m_historyList[i]);
+    }
+    return backhist;
+}
+
+wxVector<wxSharedPtr<wxWebViewHistoryItem> > wxWebViewEdge::GetForwardHistory()
+{
+    wxVector<wxSharedPtr<wxWebViewHistoryItem> > forwardhist;
+    //As we don't have std::copy or an iterator constructor in the wxwidgets
+    //native vector we construct it by hand
+    for (int i = m_historyPosition + 1; i < static_cast<int>(m_historyList.size()); i++)
+    {
+        forwardhist.push_back(m_historyList[i]);
+    }
+    return forwardhist;
+}
+
+bool wxWebViewEdge::CanGoForward() const
+{
+    if (m_historyEnabled)
+        return m_historyPosition != static_cast<int>(m_historyList.size()) - 1;
+    else
+        return false;
+
+    /*
+        BOOL result = false;
+        if (m_webView && SUCCEEDED(m_webView->get_CanGoForward(&result)))
+            return result;
+        else
+            return false;
+    */
+}
+
+bool wxWebViewEdge::CanGoBack() const
+{
+    if (m_historyEnabled)
+        return m_historyPosition > 0;
+    else
+        return false;
+    /*
+        BOOL result = false;
+
+        if (m_webView && SUCCEEDED(m_webView->get_CanGoBack(&result)))
+            return result;
+        else
+            return false;
+    */
+}
+
+void wxWebViewEdge::GoBack()
+{
+    LoadHistoryItem(m_historyList[m_historyPosition - 1]);
+    /* if (m_webView)
+        m_webView->GoBack(); */
+}
+
+void wxWebViewEdge::GoForward()
+{
+    LoadHistoryItem(m_historyList[m_historyPosition + 1]);
+    /* if (m_webView)
+        m_webView->GoForward(); */
+}
+
+void wxWebViewEdge::ClearHistory()
+{
+    m_historyList.clear();
+    m_historyPosition = -1;
+}
+
+void wxWebViewEdge::EnableHistory(bool enable)
+{
+    m_historyEnabled = enable;
+    m_historyList.clear();
+    m_historyPosition = -1;
+}
+
+void wxWebViewEdge::Stop()
+{
+    if (m_webView)
+        m_webView->Stop();
+}
+
+void wxWebViewEdge::Reload(wxWebViewReloadFlags WXUNUSED(flags))
+{
+    if (m_webView)
+        m_webView->Reload();
+}
+
+wxString wxWebViewEdge::GetPageSource() const
+{
+    return wxString();
+}
+
+wxString wxWebViewEdge::GetPageText() const
+{
+    return wxString();
+}
+
+bool wxWebViewEdge::IsBusy() const
+{
+    return m_isBusy;
+}
+
+wxString wxWebViewEdge::GetCurrentURL() const
+{
+    LPWSTR uri;
+    if (m_webView && SUCCEEDED(m_webView->get_Source(&uri)))
+        return wxString(uri);
+    else
+        return wxString();
+}
+
+wxString wxWebViewEdge::GetCurrentTitle() const
+{
+    PWSTR _title;
```

Comment:
Use of underscore prefix is really unusual, why? Could we rename it to just `title` or does this clash with something (what?).